### ISSUE TITLE
Elementwise Equal/Greater/Less not work with implicit batch

### ISSUE THREAD

**Issue by weimingzha0** [NVIDIA/TensorRT#1830]
## Environment

**TensorRT Version**:  8.2.1.8
**NVIDIA GPU**: V100
**NVIDIA Driver Version**: 460.80
**CUDA Version**: 11.2
**CUDNN Version**: 
**Operating System**:  Ubuntu 18
**Python Version (if applicable)**: 
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**: 

## Steps To Reproduce

bool_test.cc:
```c++
 % cat bool_test.cc
#include <NvInfer.h>

#include <array>
#include <iostream>

using namespace nvinfer1;
class Logger : public ILogger {
 public:
  void log(ILogger::Severity severity, const char* msg) noexcept override {
    std::cerr << "LOG: " << msg << "\n";
  }
};

static std::array<float, 4> WeightData{1, 2, 3, 4};
int main() {
  Logger logger;
  constexpr size_t MAX_WSS{1024UL * 1024UL * 1024UL * 2UL};
  auto builder = createInferBuilder(logger);
  builder->setMaxBatchSize(10);
  NetworkDefinitionCreationFlags flags = 0;
  flags = 1U << static_cast<uint32_t>(
              NetworkDefinitionCreationFlag::kEXPLICIT_BATCH);

  auto network = builder->createNetworkV2(flags);
  Dims2 shape(2, 2);
  DataType dt = DataType::kFLOAT;
  auto input = network->addInput("input", dt, shape);

  // auto input2 = network->addInput("input2", dt, shape);
  Weights weight{.type = dt, .values = WeightData.data(), .count = 4};
  auto input2 = network->addConstant(shape, weight)->getOutput(0);

  auto bin =
      network->addElementWise(*input, *input2, ElementWiseOperation::kEQUAL)
          ->getOutput(0);
  bin->setType(DataType::kBOOL);


  network->markOutput(*bin);
  auto builder_cfg = builder->createBuilderConfig();
  builder_cfg->setMaxWorkspaceSize(MAX_WSS);
  auto engine = builder->buildEngineWithConfig(*network, *builder_cfg);
}
```

To run:
```bash
g++ bool_test.cc -std=c++14 -I/usr/local/cuda-11.4/targets/x86_64-linux/include -lnvinfer -g && ./a.out
```

Error:

 [1]: 2: [elementwiseBuilder.cpp::addSupportedFormats::591] Error Code 2: Internal Error (Assertion kCudaElementWiseSupportsBool failed. Cuda backend does not support kEQUAL/kGREATER/kLESS)

Layer(Myelin): {ForeignNode[(Unnamed Layer* 0) [Constant]...(Unnamed Layer* 1) [ElementWise]]}, Tactic: 0, input[Float(2,2)] -> (Unnamed Layer* 1) [ElementWise]_output[Bool(2,2)]

**Comment by weimingzha0** [comment]
Release notes says the issue was fixed:
"Boolean input/output tensors were only supported when using explicit batch dimensions. This has been fixed in this release."
https://docs.nvidia.com/deeplearning/tensorrt/release-notes/tensorrt-8.html#rel-8-2-1

but I still see the error

**Comment by nvpohanh** [comment]
@weimingzha0 Is there any reason why you still have to use implicit batch dim? Implicit batch dim mode is deprecated and will be remove in the future. Thanks

**Comment by nvpohanh** [comment]
closing for now due to >14 days with no response. Please feel free to reopen if the issue still exists. Thanks
